accessible/field-segment.component.js

```diff
@@ -144,7 +150,12 @@ blocklyApp.FieldSegmentComponent = ng.core.Component({
   // Confirm a selection for dropdown fields.
   selectOption: function() {
     if (this.optionValue == Blockly.Msg.RENAME_VARIABLE) {
```

Comment:
Ahh, thanks! That was what I was missing -- I totally forgot about translations being an issue here.

Maybe a possible solution is to have the option value be the translation id (which would remain invariant under the user-selected language), and the option text be the actual translated text the user sees. Not sure how feasible this is though.